### ISSUE TITLE
module-import-not-at-top-of-file (E402) - missing support for site.addsitedir()

### ISSUE THREAD

**Issue by Sarcasm** [astral-sh/ruff#16247]
### Description

Failing file `ruff-E402-addsitedir.py`:

```python
#!/usr/bin/env python

import os
import site
import sys
import sysconfig

site.addsitedir(
    os.path.join(
        os.path.dirname(os.path.dirname(__file__)),
        sysconfig.get_path("purelib", vars={"base": "."}),
    )
)

from mypkg.__main__ import main

if __name__ == "__main__":
    sys.argv[0] = sys.argv[0].removesuffix(".py")
    sys.exit(main())
```

Ruff check complains:

```console
$ ruff check --select E402 ruff-E402-addsitedir.py
ruff-E402-addsitedir.py:15:1: E402 Module level import not at top of file
   |
13 | )
14 | 
15 | from mypkg.__main__ import main
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
16 | 
17 | if __name__ == "__main__":
   |

Found 1 error.
```

Looking at the code, `sys.path` and some others cases are handled

https://github.com/astral-sh/ruff/blob/0.9.6/crates/ruff_linter/src/checkers/ast/mod.rs#L552-L562
https://github.com/astral-sh/ruff/blob/524cf6e5155066132da772b9f84e2e6695f241b8/crates/ruff_linter/src/checkers/ast/mod.rs#L552-L562

but `site.addsitedir` seems missing, although it appends to `sys.path` internally:

https://github.com/python/cpython/blob/3.13/Lib/site.py#L238

BTW, thanks for Ruff, it's a great piece of software!

**Comment by VascoSch92** [comment]
I can do the fix to handle this case if needed ;-)

**Comment by VascoSch92** [comment]
I think this can be closed ;-)

**Comment by Sarcasm** [comment]
Indeed, Github did not notified me automatically of the Pull Request.

Thank you for fixing this so quickly!
